### PR TITLE
[STORM-3949] libthrift 0.18.1

### PR DESCRIPTION
**Pull request by rzo1**
## What is the purpose of the change

- Upgrade `libthrift` to 0.18.1

## How was the change tested

- Run tests locally similar to GH actions
- GH actions

### CODE REVIEW

File: DEPENDENCY-LICENSES [hunk 1]
```diff
@@ -114,6 +114,7 @@ List of third-party dependencies grouped by their license type.
         * Apache HttpCore (org.apache.httpcomponents:httpcore:4.4.10 - http://hc.apache.org/httpcomponents-core-ga)
         * Apache HttpCore (org.apache.httpcomponents:httpcore:4.4.13 - http://hc.apache.org/httpcomponents-core-ga)
         * Apache HttpCore (org.apache.httpcomponents:httpcore:4.4.1 - http://hc.apache.org/httpcomponents-core-ga)
+        * Apache HttpCore (org.apache.httpcomponents:httpcore:4.4.4 - http://hc.apache.org/httpcomponents-core-ga)
```
Comment (bipinprasad) [inline DEPENDENCY-LICENSES:117]: if possible, can we use just the 4.4.5 version below?

**Review by bipinprasad — APPROVED** [review]

File: DEPENDENCY-LICENSES (hunk 1 above)
Comment (rzo1) [reply DEPENDENCY-LICENSES]: I cleaned it up, so we only have one version of httpclient / httpcore in the project.

**Comment by agresch** [comment]
When storm upgraded to thrift 0.13 initially, we would see huge CPU spikes on supervisors and nimbus after approximately 24 hours for unknown reasons.  The problem went away when we backed out that change.  Just wondering if anyone has seen any performance issues of note with this change so far.

**Comment by rzo1** [comment]
@agresch think we will see after a 2.6.0 ist available because I doubt anyone is using the Last SNAPSHOT build in a staging environment ,et.

**Comment by avermeer** [comment]
I have been running my pre-production topologies with Storm 2.6.0 SNAPSHOT
build of the 2023/11/03, with a flow ~1000 tuples/minutes since 2 days,and
so far my metrics on latency, Kafka lag, etc aren't showing regressions ;
consumption is even more stable than with Storm 2.5.0.

on Supervisor machines, CPU & Memory are behaving "just as usual"

Please note that I haven't rebuilt my topologies bigjars with 2.6.0
dependencies (but I fixed some trivial issues related to old bad practices).

Alexandre


Le lun. 6 nov. 2023 à 22:38, Richard Zowalla ***@***.***> a
écrit :

**Pull request merged**